partiql-types/src/main/java/org/partiql/value/PartiQLValueLoader.java

```diff
@@ -0,0 +1,30 @@
+package org.partiql.value;
+
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Provides functions for loading {@link PartiQLData} into {@link PartiQLValue} instances.
+ *
+ * @see PartiQLData
+ */
+public interface PartiQLValueLoader {
+    /**
+     * Loads un-materialized {@link PartiQLData} into a materialized {@link PartiQLValue}. The {@link PartiQLData} cursor
+     * must be set <b>before</b> the value that you'd like to load.
+     * <p>
+     * This method will invoke {@link PartiQLData#next()}. This method will <b>not</b> throw an error is there is
+     * more data to be processed after the value immediately following the cursor.
+     *
+     * @param data the PartiQL data to load.
+     * @return a materialized, in-memory instance of a {@link PartiQLValue} containing the contents of the {@code data}.
+     */
+    @NotNull
+    PartiQLValue loadSingleValue(@NotNull PartiQLData data);
```

Comment:
Why single value? Would we ever have an Iterator<PartiQLValue> rather than a CollectionValue. Should just be `load`